sklearn/callback/_progressbar.py

```diff
@@ -24,15 +24,41 @@ def __init__(self, max_estimator_depth=1):
 
         self.max_estimator_depth = max_estimator_depth
 
-    def _on_fit_begin(self, estimator):
+    def on_fit_begin(self, estimator):
+        """Callback hook called at the beginning of the estimator's fit method.
+
+        Parameters
+        ----------
+        estimator : estimator instance
+            Estimator the callback is attached to.
+        """
```

Comment:
Let's tweak the `sklearn.tests.test_docstrings` test to ignore these methods. It can be done in `get_all_methods`. We can add something like
```py
if isinstance(Klass, Callback) and name in ("on_fit_begin", "on_fit_end", "on_fit_task_end"):
    continue
```